components/shared/compositing/display_list.rs

```diff
@@ -106,6 +111,74 @@ pub struct ScrollableNodeInfo {
     pub offset: LayoutVector2D,
 }
 
+impl ScrollableNodeInfo {
+    fn scroll_to_offset(
+        &mut self,
+        new_offset: LayoutVector2D,
+        context: ScrollType,
+    ) -> Option<LayoutVector2D> {
+        if !self.scroll_sensitivity.x.contains(context) &&
+            !self.scroll_sensitivity.y.contains(context)
+        {
+            return None;
```

Comment:
It seems inconsistent that if the element isn't scrollable in any axis this returns None, but if it's only scrollable in a single axis then it returns `Some(LayoutVector2D)`. Should it be `Vector2D<Option<f32>>`?

`scroll_node_or_ancestor` relies on the current logic but it seems a bit broken to me, e.g. the request is for scrolling to a position with a different vertical offset, and the scrolling container can only scroll horizontally, then this will return a Some(), and `scroll_node_or_ancestor` will not keep iterating ancestors. Is that correct?

Not blocking this PR since it's an existing thing.

Reply:
Yes, I think your understanding of the behavior is correct. We would need to return a more complicated data structure if it's possible to scroll more than a single node with the same operation. This cannot happen via script and I'm not exactly sure how things should operate when processing input events. Likely we really only want to support scrolling a single scroll node at a time, but I could be wrong.